tasmota/tasmota_support/support.ino

```diff
@@ -590,6 +590,33 @@ char* Trim(char* p) {
   return p;
 }
 
+size_t HexStringToBinaryString(char** source) {
+  size_t str_len = strlen(*source+2);
+  char *s = *source+2;
+  char *new_str = (char *)malloc(str_len/2);
+  for (int i = 0; i < str_len; i++) {
+        unsigned char val;
+        char cur = s[i];
+        if (cur >= 97) {
+            val = cur - 97 + 10;
```

Comment:
`val` could have a much larger value than expected. Can you add a guarding condition, and decide what value you put for illegal charachters? `0` I guess?

Reply:
Added check to ensure that all values in string contain 0-9, a-z or A-Z